File: 2023/day1/part1/solution.swift
import Foundation

let digits: String = "0123456789"
let fileURL: URL = FileManager.default.homeDirectoryForCurrentUser
    .appendingPathComponent("Desktop/advent-of-code/2023/day1/part1/input.txt")
let text: String = try! String(contentsOf: fileURL, encoding: .utf8)

var result: Int = 0
for line in text.split(separator: "\n") {
    var digs: [Character] = line.filter { digits.contains($0) }

    switch digs.count {
    case 1: result += Int(String(repeating: digs[0], count: 2))!
    default: result += Int(String([digs[0], digs[digs.count - 1]]))!
    }
}

print(result)

File: 2023/day1/part1/solution.py
from string import digits

FILE_PATH: str = 'input.txt'
with open(FILE_PATH, 'r') as input:
    lines: list[str] = input.readlines()

result: int = 0
for line in lines:
    match [char for char in line if char in digits]:
        case [a]:
            result += int(a + a)
        case [a, *_, b]:
            result += int(a + b)

print(result)
